### PR TITLE
Added search

### PR DESCRIPTION
**Pull request by person-m**
Added some basic filters (name, category)

### CODE REVIEW

File: src/routes/pull.rs [hunk 1]
```diff
@@ -9,8 +9,26 @@ use toml;
 use manifest::Manifest;
 use RAVEN_REPOSITORY_PATH;
 
+#[derive(Debug, FromForm)]
```
Comment (Arignir) [inline src/routes/pull.rs]: Your code should follow (at the best you can) Rust's API guidelines:

https://rust-lang-nursery.github.io/api-guidelines/

Even though those are guidelines for the std library, they are also good guidelines for any Rust project.

Especially, you missed the `C-COMMON-TRAITS` guideline.
It's not a real problem as this is a binary and not a library, but you're new to Rust so you might want to have good habits right now even when it's not the most relevant.

File: src/routes/pull.rs [hunk 2]
```diff
@@ -27,3 +45,31 @@ fn pull() -> Result<Json<Vec<Manifest>>, Error> {
...
+        let mut s = file
+            .metadata()
+            .map(|m| String::with_capacity(m.len() as usize))
+            .unwrap_or_default();
+
+        file.read_to_string(&mut s)?;
+        manifests.push(toml::from_str(&s)?);
+    }
+
+    if let Some(p) = filter {
```
Comment (Arignir) [inline src/routes/pull.rs]: `p` is not a good name. It's not even the abbreviation of `ManifestFilter`. 

(Same for `n` and `c` later)

File: src/routes/pull.rs (hunk 1 above)
Comment (person-m) [reply src/routes/pull.rs]: Could you be a bit more specific for this case please?

Comment (Arignir) [reply src/routes/pull.rs]: The given guidelines is a long set of rules to follow in order to write idiomatic Rust code.

You have a checklist (to use before commit) [here](https://rust-lang-nursery.github.io/api-guidelines/checklist.html).

One of this rule is labeled `C-COMMON-TRAITS` (all rules have a label like this, so they can be referenced easily).

More information about this rule can be find [here](https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#types-eagerly-implement-common-traits-c-common-traits).

It basically boils down to "use `#[derive]` on as much standard trait as you can". You'll want to derive Serde traits too, if possible (`Serialize`/`Deserialize`).

**Pull request merged**